Clicketing

Thechi2000 opened the pull request:
> [!WARNING] 
> This is a PR hijacking, please remain seated and calm while we redirect it to 11/02.

Adds the data models for ICBD, as well as most of the ones for the other events (see below). ICBD flows will come later on.


Previous description: 
======================

Add the data models and flows for the Clicketing app.

#### Models

- [x] Event
  - [x] Shared
    - [x] Max registration count
  - [x] Hello World
  - [x] Faculty dinner
- [ ] Registration
  - [x] Shared
  - [ ] Hello World
  - [x] Faculty dinner
  - [x] Staffing

#### Flows
- [ ] Shared
  - [x] Send registration validation mail
  - [ ] Generate staffing sheet
- [ ] Hello World
  - [ ] Generate teams sheet
- [x] Faculty dinner
  - [x] Generate meals count

Thechi2000 commented:
It's still missing the Hello world specific sections, but this can be implemented later imo.
@SidonieBouthors @NoeTerrier

Thechi2000 commented:
(funny how this PR got no reviews in 4 months :eyes:)

@SidonieBouthors @NoeTerrier please check the ICBD data models. I'll make another PR once I'll have implemented the flows in directus (as well as one on infra to automatically deploy those).

SidonieBouthors commented:
I've had a look at the ICBD models. Overall it looks good to me but I have a quick question
As far as I can tell there is no activity type, but from the example content it looks like maybe the idea is that all activities of a certain type are grouped together (for example, all mock coding interviews) with several hosts and timeslots. I think this is ok for some type of activities, but for Talks it would be nice to know the speaker for a specific talk, and have a dedicated description etc, while also having the possibility to fetch all talks from the db. The simplest way to do this would be to add an activity type that can be shared by several activities, what do you think ?

Thechi2000 commented:
I've added an activity type field (dropdown between "Talk", "Discussion" and "Other").

I've reworked a bit how the rest work, here is a walk-through of the workflow I'm thinking about:

## Activities registrations

The table `icbd_activities_registration` (`icbd_activities >=< registrations`) maps each registration with one or more activities. Each of these entry also holds the start time of the timeslot the user is registered to, and whether they attended the activity. For talks and discussions, the time and attended are automatically set (activity start time and true), while for SN and interviews, it will be filled manually upon payment of the deposit.

I also added a `max_attendees` field to each timeslots of the activities. When set to 0, we consider that there is no limit, which allows to easily distinguish between SN/interviews and talks/discussions.

## Deposit management

For each activity registration, we keep track of whether it was attended or not. Once all required activities are attended, then the user can retrieve their deposit. Here, a flow should update the `can_retrieve_deposit` field to keep track of this.

SidonieBouthors commented:
Sounds good :)

NoeTerrier reviewed (approved):
Nice changes, I updated a bit the preview of the collections, but it's really cool.
